fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3029,6 +3033,8 @@ class DDTeamCollectionImpl {
 					// Update the wigglingId only if it matches the locality.
 					if (server->getLastKnownInterface().locality.get(localityKey.get()) == localityValue) {
 						self->wigglingId = res.begin()->first;
+					} else {
+						wait(self->eraseStorageWiggleMap(&metadataMap, res.begin()->first));
```

Comment:
The erase and read are not in the same transaction. This won't have the atomic guarantee

Reply:
I was thinking the case

- readStorageWiggleMap first read the value in transaction A, and find its locality is bad
- Another wiggler might update the map with a valid ID in transaction B
- readStorageWiggleMap clear the value written by transaction B based on the value read in transaction A

So probably now this is not a problem, give that there’s only one alive perpetual wiggler